interfaces/builtin/media_control.go

```diff
@@ -0,0 +1,59 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+// See https://www.kernel.org/doc/Documentation/userspace-api/media/mediactl/media-controller-intro.rst
+const mediaControlSummary = `allows access to media control devices`
+
+// The kernel media controller allows connecting and configuring
+// media hardware subsystems
+const mediaControlBaseDeclarationSlots = `
+  media-control:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const mediaControlConnectedPlugAppArmor = `
+# Control of media devices
+/dev/media[0-9]* rw,
+
+# Access to V4L subnodes configuration
+# See https://www.kernel.org/doc/html/v4.12/media/uapi/v4l/dev-subdev.html
+/dev/v4l-subdev[0-9]* rw,
+`
+
+var mediaControlConnectedPlugUDev = []string{
+	`SUBSYSTEM=="media", KERNEL=="media[0-9]*"`,
+	`SUBSYSTEM=="video4linux", KERNEL=="v4l-subdev[0-9]*"`,
+}
```

Comment:
FTR:
```
ubuntu@ubuntu:~$ udevadm info --attribute-walk --name=/dev/media0
  looking at device '/devices/platform/amba_pl@0/amba_pl@0:vcap_hdmi_input_v_proc_ss_0/media0':
    KERNEL=="media0"
    SUBSYSTEM=="media"
    DRIVER==""
    ATTR{model}=="Xilinx Video Composite Device"
ubuntu@ubuntu:~$ udevadm info --attribute-walk --name=/dev/v4l-subdev0
  looking at device '/devices/platform/amba_pl@0/a0040000.v_proc_ss/video4linux/v4l-subdev0':
    KERNEL=="v4l-subdev0"
    SUBSYSTEM=="video4linux"
    DRIVER==""
    ATTR{index}=="1"
    ATTR{dev_debug}=="0"
    ATTR{name}=="a0040000.v_proc_ss"
ubuntu@ubuntu:~$ udevadm info --attribute-walk --name=/dev/v4l-subdev1
  looking at device '/devices/platform/amba_pl@0/a0000000.v_hdmi_rx_ss/video4linux/v4l-subdev1':
    KERNEL=="v4l-subdev1"
    SUBSYSTEM=="video4linux"
    DRIVER==""
    ATTR{dev_debug}=="0"
    ATTR{index}=="2"
    ATTR{name}=="a0000000.v_hdmi_rx_ss"
```

Reply:
the rules look fine to me, thanks for posting this